### PR TITLE
WIP: Add settings add chem comp

### PR DESCRIPTION
**Pull request by RiesBen**
Hi, 
here i try to add an initial draft of the chemComposition.
I also slightly modified the file structure, but this can be reverted if wished?

This is currently WIP ;)

### CODE REVIEW

File: gufe/settings/models/chemical_composition.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"""
+Pydantic models used for storing settings for chemical Composition.
+"""
+
+from .abstract import SettingsBaseModel
+from typing import List, Optional
+
+# abstract entity
+class Chemical_Entity(SettingsBaseModel):
```
Comment (dotsdl) [inline gufe/settings/models/chemical_composition.py:9]: No underscore in class name.

File: gufe/settings/models/chemical_composition.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    exists: bool = False
+
+
+# Solutes:
+class Solute(Chemical_Entity):
+    num_compounds: int = 0
+
+
+# Protein:
+class Protein_Cofactors(Chemical_Entity):
```
Comment (dotsdl) [inline gufe/settings/models/chemical_composition.py:19]: No underscore in class name.

**Review by dotsdl — COMMENTED** [review]
Thanks @RiesBen! I did a quick review on mostly style things. Can you tell me what these objects are supposed to do? It seems odd to me that we are including system composition as part of settings, but I don't know if I missed out on some discussions here.

**Review by richardjgowers — CHANGES_REQUESTED** [review]
I think this is off on a tangent a little.  `Settings` are designed to capture everything *except* this sort of information.  I think *all* of the information here will be collected in various `*Component` classes

**Comment by RiesBen** [comment]
@richardjgowers @dotsdl 
Hm I see your point, I tried to add this layer as it was defined in the taxonomy doc. 
For a simulation setup I think, it might be beneficial to have a fast look up on this, to support the reduced potential settings or other layers. But I also see, that this is represented by the Component classes. 
Therefore one could also think about removing it again? Then we should also remove it from the doc, or did I miss understand the doc?

**Comment by IAlibay** [comment]
> Then we should also remove it from the doc, or did I miss understand the doc?

My 2 cent is that I wouldn't remove them from the doc, the components are a "setting" in themselves, i.e. they define how the positions and chemistry of your system are "set".

**Pull request closed without merging**